File: frontend/src/constants/caseStatus.ts
/**
 * Case status constants
 */
export const CASE_STATUS = {
  OPEN: 'open',
  PENDING: 'pending',
  IN_PROGRESS: 'in_progress',
  COMPLETED: 'completed',
  CLOSED: 'closed',
  REJECTED: 'rejected',
  CANCELLED: 'cancelled',
} as const;

export type CaseStatus = typeof CASE_STATUS[keyof typeof CASE_STATUS];

/**
 * Cost status constants
 */
export const COST_STATUS = {
  PENDING: 'pending',
  APPROVED: 'approved',
  REJECTED: 'rejected',
} as const;

export type CostStatus = typeof COST_STATUS[keyof typeof COST_STATUS];

/**
 * Final cost status constants
 */
export const FINAL_COST_STATUS = {
  PENDING: 'pending',
  APPROVED: 'approved',
  REJECTED: 'rejected',
} as const;

export type FinalCostStatus = typeof FINAL_COST_STATUS[keyof typeof FINAL_COST_STATUS];

File: mobile_flutter/lib/shared/constants/case_status.dart
enum CaseStatus {
  open,
  pending,
  inProgress,
  completed,
  closed,
  rejected,
  cancelled,
}

extension CaseStatusExtension on CaseStatus {
  String get displayName {
    switch (this) {
      case CaseStatus.open:
        return 'Open';
      case CaseStatus.pending:
        return 'Pending';
      case CaseStatus.inProgress:
        return 'In Progress';
      case CaseStatus.completed:
        return 'Completed';
      case CaseStatus.closed:
        return 'Closed';
      case CaseStatus.rejected:
        return 'Rejected';
      case CaseStatus.cancelled:
        return 'Cancelled';
    }
  }
}

